src/event_handlers/scans.rs

```diff
@@ -156,6 +157,24 @@ impl ScanHandler {
                     self.ordered_scan_url(vec![target], ScanOrder::Initial)
                         .await?;
                 }
+                Command::ModifyScanLimit(limit) => {
+
+                    let available = self.limiter.available_permits();
+                    
+                    match limit.cmp(&available){
+                        Ordering::Greater => {
+                            self.limiter.add_permits((limit - available) as usize)
+                        }
+                        Ordering::Less => {
+                            self.limiter.acquire_many((available - limit) as u32).await?.forget();
```

Comment:
`available` is a usize, which on my system is 64 bits. Also, when `ScanHandler::new` is called, and the user didn't use `-L`, the initial limiter is set to `usize::MAX >> 4`.

So, when i kick off a scan without a limit on my machine, `available` is some number in the ballpark of 1,152,921,504,606,846,975. 

That means when we subtract some normal number, like 3, from that huge number, we're left with a number that overflows a u32. The use of the `as` keyword + the overflow results in calling `acquire_many` with the value `0` when the conditions i've described are met. 

Unfortunately, because `acquire_many` only takes a u32, it's going to effectlvely pause the scan if we try to subtract `u32::MAX` a bunch of times. The easiest solution is to change `ScanHandler::new` to call `limiter.add_permits(u32::MAX as usize);`. That way we'll always be working with less than a u32 by the time we get here.